src/Analysis/GWASWizard/shared/useDebounce.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+
+// Custom hook that will set a timeout delay on updating state.
+// Source: https://dev.to/arnonate/debouncing-react-query-with-hooks-2ek6, https://usehooks.com/useDebounce/
+export function useDebounce(value, delay) {
+    // State and setters for debounced value
+    const [debouncedValue, setDebouncedValue] = useState(value);
+    useEffect(
+      () => {
+        // Update debounced value after delay
+        const handler = setTimeout(() => {
+          setDebouncedValue(value);
+        }, delay);
+        // Cancel the timeout if value changes (also on delay change or unmount)
+        // This is how we prevent debounced value from updating if value is changed ...
+        // .. within the delay period. Timeout gets cleared and restarted.
+        return () => {
+          clearTimeout(handler);
+        };
+      },
+      [value, delay] // Only re-call effect if value or delay changes
```

Comment:
does this behave the same way when you remove `delay` from dependency array? its always 800 i think

Reply:
run `npm run eslint` if u didnt already

Reply:
good point. I'll remove that one.

Reply:
Removing the delay from the useEffect causes this eslint warning though:
```
/Users/pieterlukasse/dev/data-portal/src/Analysis/GWASWizard/shared/useDebounce.js
  23:5  warning  React Hook useEffect has a missing dependency: 'delay'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
```
...so decided to keep it.

Reply:
mkay no worries, thought so